### DISCUSSION TITLE
Optimization use MQTT TCP via weak Network

### DISCUSSION THREAD

**Discussion by Entfi** [emqx/emqx#9543] [General]
Hey, 
we try to optimize the MQTT usage over a slow RF Network. (Yes, TCP may was not the best decision for that and UDP (MQTT-SN) would fit better).
Some nodes have a very long round-trip time per packet, around 160ms - 7 seconds.
 
The issue with hat is the TCP stack try to resend a lot of retransmissions that’s lading in even more delay and high network utilization. This result in failing connection to the EMQX.

My question is who/where is the TCP behavior controlled when EMQX(4.4.11) is running in a Docker container? 
Our try to change with `sudo ip route change 172.19.0.0/16 dev eth0 rto_min 60s rtt 60s rttvar 5s` take no change for MQTT connections.

 
![TCP-issuse](https://user-images.githubusercontent.com/120551468/207552631-26d30d4d-9775-4b9d-9811-e8ce4da52208.png)

Many thaks for tipps

**Comment by zmstone** [comment]
I have not tried, but I suppose you'll need `--priviledged` for the container.
